zebra/zebra_mpls.c

```diff
@@ -197,10 +201,29 @@ static int lsp_install(struct zebra_vrf *zvrf, mpls_label_t label,
 	 */
 	for (nexthop = re->nhe->nhg.nexthop;
 	     nexthop; nexthop = nexthop->next) {
+		nexthop_via_recursive = nexthop;
 		/* Skip inactive and recursive entries. */
 		if (!CHECK_FLAG(nexthop->flags, NEXTHOP_FLAG_ACTIVE))
 			continue;
-		if (CHECK_FLAG(nexthop->flags, NEXTHOP_FLAG_RECURSIVE))
+		if (CHECK_FLAG(nexthop->flags, NEXTHOP_FLAG_RECURSIVE) &&
+		    zvrf->zebra_mpls_fec_recursive_seamless) {
+			if (nexthop->type == NEXTHOP_TYPE_IPV4 ||
+			    nexthop->type == NEXTHOP_TYPE_IPV4_IFINDEX) {
+				p.family = AF_INET;
+				memcpy(&p.u.prefix4, &nexthop->gate.ipv4,
+				       sizeof(p.u.prefix4));
+				p.prefixlen = IPV4_MAX_BITLEN;
+			} else if (nexthop->type == NEXTHOP_TYPE_IPV6 ||
+				   nexthop->type == NEXTHOP_TYPE_IPV6_IFINDEX) {
+				p.family = AF_INET6;
+				memcpy(&p.u.prefix6, &nexthop->gate.ipv6,
+				       sizeof(p.u.prefix6));
+				p.prefixlen = IPV6_MAX_BITLEN;
+			}
+
+			re_recursive = rib_lookup(&p, zvrf->vrf->vrf_id);
```

Comment:
I'm willing to bet that this 1st commit in the tree needs rib_lookup to work.  This will not even compile.  I'm not interested in taking code that will fail me when I have a git bisect.  Please take the time and make sure the code will work and compile in sequence

Reply:
I'm working on the new version that has the code which compiles in sequence. Thank you for the comment.